src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriterUtils.cs

```diff
@@ -1,43 +1,45 @@
 //---------------------------------------------------------------------
-// <copyright file="ODataBatchWriterUtils.cs" company="Microsoft">
+// <copyright file="ODataMultipartMixedBatchWriterUtils.cs" company="Microsoft">
 //      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
 // </copyright>
 //---------------------------------------------------------------------
 
-namespace Microsoft.OData
+namespace Microsoft.OData.MultipartMixed
 {
     #region Namespaces
     using System;
+    using System.Collections.Generic;
     using System.Diagnostics;
     using System.Globalization;
     using System.IO;
+    using System.Linq;
     #endregion Namespaces
 
     /// <summary>
     /// Helper methods used by the ODataBatchWriter.
     /// </summary>
-    internal static class ODataBatchWriterUtils
+    internal static class ODataMultipartMixedBatchWriterUtils
     {
         /// <summary>
         /// Creates a new batch boundary string based on a randomly created GUID.
         /// </summary>
-        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a resonse.</param>
+        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a response.</param>
         /// <returns>The newly created batch boundary as string.</returns>
         internal static string CreateBatchBoundary(bool isResponse)

```

Comment:
>CreateBatchBoundary [](start = 31, length = 19)

For consistency with CreateChangeSetBoundary, consider passing in a boundaryid that is generated when calling this function.

Reply:
I don't think the user/client can specify the boundary the same way as can be done for changeSetBoundary.
The batch boundary is created by ODataMultipartMixedBatchFormat.GetContentType().

---
In reply to: [155385370](https://github.com/OData/odata.net/pull/992#discussion_r155385370) [](ancestors = 155385370)

Reply:
Yep; I understand. I was just suggesting that, for API consistency, the caller call Guid.NewGuid() rather than generating it here. But generating it here is fine too.

---
In reply to: [155407361](https://github.com/OData/odata.net/pull/992#discussion_r155407361) [](ancestors = 155407361,155385370)